src/hooks/useTxNotifications.ts

```diff
@@ -68,6 +76,44 @@ const useTxNotifications = (): void => {
       unsubFns.forEach((unsub) => unsub())
     }
   }, [dispatch, safeAddress, chain?.shortName])
+
+  /**
+   * Show a notification when a transaction is awaiting confirmation on Safe load
+   */
+
+  const { page } = useTxQueue()
+  const pendingTxs = useAppSelector(selectPendingTxs)
+  const notifications = useAppSelector(selectNotifications)
+
+  useEffect(() => {
+    const items = page?.results?.filter(isTransactionListItem)
+
+    if (!items || items.length === 0) {
+      return
+    }
+
+    const { transaction } = items[0]
+
+    if (transaction.txStatus !== TransactionStatus.AWAITING_CONFIRMATIONS || pendingTxs[transaction.id]) {
+      return
+    }
```

Comment:
I would extract this part out into a function called `getFirstQueuedTransaction`.

Reply:
I've extracted it to a memoised variable because it would otherwise have a lot of arguments with the new checks. What do you think?

Reply:
Sure, as long as it's encapsulated 👍